client/main.lua

```diff
@@ -1,44 +1,207 @@
+                marker.color.r, marker.color.g, marker.color.b, marker.color.a, -- color RBGA
+                false, true, 2, false, nil, nil, false
+                )
+
+                if self.currentDistance < 1 then
+                    SetTextComponentFormat("STRING")
+                    AddTextComponentString(Lang:t('interiorZones.logout'))
+                    DisplayHelpTextFromStringLabel(0, 0, 1, 20000)
+                    if IsControlJustPressed(0, 38) then
+                        TriggerEvent('qbx-properties:server:Logout')
+                    end
+                end
+            end
+        end
+    end
+end
+
+--- Get the rounded coords
+---@param coords table
+---@return table
+function GetRoundedCoords(coords)
+    local newcoords = {}
+    for k, v in pairs(coords) do
+        newcoords[k] = math.floor(v*1000)/1000
+    end
+    return newcoords
+end
+
+--- Create a blip for the Apartment
+---@param coords vector3
+---@param name string name that will be displayed on the blip
+function AddApartmentBlip(coords, name)
+    AddBlip('apartment', name, coords, Config.Apartments.Blip.sprite, Config.Apartments.Blip.color, Config.Apartments.Blip.scale)
+end
+
+--- Create a blip for the property/apartment
+---@param blipId integer | string property id
+---@param name string name that will be displayed on the blip
+---@param coords vector3
+---@param blip? integer
+---@param color? integer
+---@param size? integer
+function AddBlip(blipId, name, coords, blip, color, size)
+    if not blipId or not coords or not blip or not color or not size or not name then return end
+    blips[blipId] = AddBlipForCoord(coords.x, coords.y, coords.z)
+    SetBlipSprite(blips[blipId], blip)
+    SetBlipDisplay(blips[blipId], 4)
+    SetBlipScale(blips[blipId], size)
+    SetBlipColour(blips[blipId], color)
+    SetBlipAsShortRange(blips[blipId], true)
+    BeginTextCommandSetBlipName("STRING")
+    AddTextComponentString(name)
+    EndTextCommandSetBlipName(blips[blipId])
+end
+
+--- Remove blips
+function RemoveBlips()
+    if not next(blips) then return end
+    for _, blip in pairs(blips) do
+        RemoveBlip(blip)
+    end
+    blips = {}
+end
+--#endregion Functions
+
+---@param players table
+---@param conceal boolean
+local function concealPlayers(players, conceal)
+    if GetInvokingResource() ~= nil then return end
+    if not players then return end
+
+    for _, player in ipairs(players) do
+        NetworkConcealPlayer(player, conceal, false)
+    end
+end
+
+local function concealVehicles(netids, conceal)
+    if GetInvokingResource() ~= nil then return end
+    if not netids then return end
+
+    for _, netid in ipairs(netids) do
+        local entity = NetToVeh(netid)
+        NetworkConcealEntity(entity, conceal)
+    end
+end
+
+RegisterNetEvent('qbx-property:client:concealPlayers', concealPlayers)
+RegisterNetEvent('qbx-property:client:concealEntities', concealVehicles)
```

Comment:
When do we conceal players or entities?

Reply:
entities would be for decoration props and/or vehicles